scripts/fund-round-creator.sh

```diff
@@ -0,0 +1,373 @@
+#!/bin/bash
+
+set -e  # Exit on errors
+
+# Define directories
+FUND_ROUND_DIR="fund-round"
+APPS_DIR="apps"
+BRANCHES_CREATED=()  # Array to track created branches
+
+print_header() {
+    echo -e "\n\e[1;34m$1\e[0m"
+}
+
+print_prompt() {
+    echo -e -n "\e[1;33m$1\e[0m"
+}
+
+print_message() {
+    echo -e "\e[1;37m$1\e[0m"
+}
+
+print_error() {
+    echo -e "\e[1;31m$1\e[0m"
+}
+
+print_end() {
+    echo -e "\n\e[1;36m$1\e[0m"
+}
+
+# Function to cleanup branches
+cleanup_branches() {
+    echo -e "\n\e[1;31mAn error occurred. Cleaning up created branches...\e[0m"
+    for branch_info in "${BRANCHES_CREATED[@]}"; do
+        repo_path=$(echo "$branch_info" | cut -d '|' -f 1)
+        branch_name=$(echo "$branch_info" | cut -d '|' -f 2)
+        echo "Deleting branch $branch_name in $repo_path..."
+        cd "$repo_path"
+        git checkout main  # Fallback to main/master
+        git branch -D "$branch_name" || true  # Force delete branch if it exists
+        cd - >/dev/null
+    done
+}
+
+# Trap to handle script errors
+trap cleanup_branches ERR
+
+create_git_branch() {
+    local repo_path=$1
+    local branch_name=$2
+
+    cd "$repo_path"
+
+    # Check if branch already exists
+    if git branch --list $branch_name | grep -q $branch_name; then
+        print_message "Branch $branch_name already exists in $repo_path. Deleting existing branch..."
+        git checkout main
+        git branch -D "$branch_name"
+        git checkout -b "$branch_name"
+    else
+        print_message "Creating and switching to new branch $branch_name in $repo_path..."
+        git checkout -b "$branch_name"
+    fi
+
+    BRANCHES_CREATED+=("$repo_path|$branch_name")  # Track created branch
+
+    cd - >/dev/null
+}
+
+commit_changes() {
+    local repo_path=$1
+    local commit_message=$2
+
+    cd "$repo_path"
+
+    print_message "Git status:"
+    git status -u
+    print_message "Git diff:"
+    git --no-pager diff
+    git add .
+    git commit --no-verify -m "$commit_message"
+
+    cd - >/dev/null
+}
+
+
+###############################################################################
+#   Main script
+###############################################################################
+
+# List all directories inside the fund-round directory
+print_header "Available directories in $FUND_ROUND_DIR:"
+dirs=()
+i=1
+for dir in "$FUND_ROUND_DIR"/*; do
+    if [[ -d "$dir" ]]; then
+        dirs+=("$dir")
```

Comment:
```suggestion
for dir in $(find $FUND_ROUND_DIR -type d -depth 1); do
    dirs+=("$dir")
```

Reply:
Your suggestion didn't work. Using `find`, this worked instead:
`find $FUND_ROUND_DIR -mindepth 1 -maxdepth 1 -type d | sed 's|.*/||'`

Reply:
Yeah ok compat issues maybe. No biggie.